packages/data-table-server/src/dataTable/internal/AnalyticsDataTable.ts

```diff
@@ -0,0 +1,80 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2022 Beyond Essential Systems Pty Ltd
+ */
+
+import { Aggregator } from '@tupaia/aggregator';
+import { TupaiaApiClient } from '@tupaia/api-client';
+import { DataBroker } from '@tupaia/data-broker';
+import { yup, yupUtils, hasNoContent, takesDateForm } from '@tupaia/utils';
+import { AbstractDataTable } from '../AbstractDataTable';
+
+const paramsSchema = yup.object().shape({
+  hierarchy: yup.string().required(),
+  dataElementCodes: yup.array().of(yup.string().required()).strict().required(),
+  organisationUnitCodes: yup.array().of(yup.string().required()).strict().required(),
+  startDate: yup
+    .string()
+    .test(
+      yupUtils.yupTestAny([hasNoContent, takesDateForm], 'startDate should be in ISO 8601 format'),
+    ),
+  endDate: yup
+    .string()
+    .test(
+      yupUtils.yupTestAny([hasNoContent, takesDateForm], 'endDate should be in ISO 8601 format'),
+    ),
+  aggregations: yup.array().of(
+    yup.object().shape({
+      type: yup.string().required(),
+      config: yup.object(),
+    }),
+  ),
+});
+
+const configSchema = yup.object();
+
+type Analytic = { period: string; organisationUnit: string; dataElement: string; value: unknown };
+
+export class AnalyticsDataTable extends AbstractDataTable<
+  typeof paramsSchema,
+  typeof configSchema,
+  Analytic
+> {
+  public constructor(apiClient: TupaiaApiClient, config: unknown) {
+    super(paramsSchema, configSchema, apiClient, config);
+  }
+
+  protected async safelyFetchData(params: {
+    hierarchy: string;
+    dataElementCodes: string[];
+    organisationUnitCodes: string[];
+    startDate?: string;
+    endDate?: string;
+    aggregations?: { type: string; config?: Record<string, unknown> }[];
+  }) {
+    const {
+      hierarchy,
+      dataElementCodes,
+      organisationUnitCodes,
+      startDate,
+      endDate,
+      aggregations,
+    } = params;
+
+    const aggregator = new Aggregator(new DataBroker({ services: this.apiClient }));
```

Comment:
No comment here just curious, do we need an Aggregator? I remember you saying something about not needing it so much?

Reply:
I'd love to bypass it ultimately, but in the short term I feel it's necessary. As it stands, we can fairly easily convert all existing reports to using data-tables if we just take their `fetch` and `aggregate` configs and pass them in to this data-table. But if we wanted to skip `Aggregator`, then we'd need to take the time to re-write each report to handle the stuff currently being done in `Aggregator` manually. It's possible, but would be a large cleanup task.

So for now lets use it, and then clean things up as we go?